var/spack/repos/builtin/packages/sz/package.py

```diff
@@ -32,17 +30,73 @@ class Sz(AutotoolsPackage):
     version('1.4.9.2',  sha256='9dc785274d068d04c2836955fc93518a9797bfd409b46fea5733294b7c7c18f8')
 
-    variant('fortran', default=False,
-            description='Enable fortran compilation')
+
+    variant('python', default=False, description="builds the python wrapper")
+    variant('netcdf', default=False, description="build the netcdf reader")
+    variant('hdf5', default=False, description="build the hdf5 filter")
+    variant('pastri', default=False, description="build the pastri mode")
+    variant('time_compression', default=False, description="build the time based compression mode")
+    variant('random_access', default=False, description="build the random access compression mode")
+    variant('fortran', default=False, description='Enable fortran compilation')
+    variant('shared', default=True, description="build shared versions of the libraries")
 
     # Part of latest sources don't support -O3 optimization
     # with Fujitsu compiler.
     patch('fix_optimization.patch', when='@2.0.2.0:%fj')
 
-    def configure_args(self):
+
+    depends_on('zlib')
+    depends_on('zstd')
+
+    extends('python', when="+python")
+    depends_on('python@3:', when="+python")
+    depends_on('swig@3.12:', when="+python")
+    depends_on('py-numpy', when="+python")
+    depends_on('hdf5', when="+hdf5")
+    depends_on('netcdf-c', when="+netcdf")
+
+    #before 2.1.8.1 this was a autotools package
+    @when("@:2.1.8.0")
+    def build(self, spec, prefix):
+        pass
+
+    #before 2.1.8.1 this was a autotools package
+    @when("@:2.1.8.0")
+    def cmake(self, spec, prefix):
+        pass
+
+    #before 2.1.8.1 this was a autotools package
+    @when("@:2.1.8.0")
+    def install(self, spec, prefix):
+        configure_args = ["--prefix=" + prefix]
+        if "+fortran" in spec:
+            configure_args.append("--enable-fortran")
+        else:
+            configure_args.append("--disable-fortran")
+        configure(*configure_args)
+        make()
+        make("install")
+        
+
+    def cmake_args(self):
         args = []
-        if '+fortran' in self.spec:
-            args += ['--enable-fortran']
+        if "+python" in  self.spec:
+            args.append("-DBUILD_PYTHON_WRAPPER=ON")
+        if "+netcdf" in  self.spec:
+            args.append("-DBUILD_NETCDF_READER=ON")
+        if "+hdf5" in  self.spec:
+            args.append("-DBUILD_HDF5_FILTER=ON")
+        if "+pastri" in  self.spec:
+            args.append("-DBUILD_PASTRI=ON")
+        if "+time_compression" in  self.spec:
+            args.append("-DBUILD_TIMECMPR=ON")
+        if "+random_access" in  self.spec:
+            args.append("-DBUILD_RANDOMACCESS=ON")
+        if "+fortran" in self.spec:
+            args.append("-DBUILD_FORTRAN=ON")
+        if "+shared" in  self.spec:
+            args.append("-DBUILD_SHARED_LIBS=ON")
         else:
-            args += ['--disable-fortran']
+            args.append("-DBUILD_SHARED_LIBS=OFF")
```

Comment:
For all of these variants, can you explicitly disable them in an else clause? CMake likes to automatically enable things depending on if their dependencies are installed on the OS, leading to non-deterministic builds with missing dependencies.

Reply:
I wrote the CMake file for SZ, so I am confident that it doesn't do that, but I can add the explicit off statements you request.